packages/vuetify/src/components/VOverlay/VOverlay.tsx

```diff
@@ -0,0 +1,278 @@
+  'global', // specific viewport position, usually centered
+  'connected', // connected to a certain element
+  'flexible', // connected to an element with the ability to overflow or shift if it doesn't fit in the screen
+] as const
+
+const scrollStrategies = [
+  'close',
+  'block',
+  'reposition',
+] as const
+
+interface ScrimProps {
+  [key: string]: unknown
+  modelValue: boolean
+  color: BackgroundColorData
+}
+function Scrim (props: ScrimProps) {
+  const { modelValue, color, ...rest } = props
+  return (
+    <Transition name="fade-transition" appear>
+      { props.modelValue && (
+        <div
+          class={[
+            'v-overlay__scrim',
+            props.color.backgroundColorClasses.value,
+          ]}
+          style={ props.color.backgroundColorStyles.value }
+          { ...rest }
+        />
+      )}
+    </Transition>
+  )
+}
+
+interface ScrollStrategy {
+  enable (): void
+  disable (): void
+}
+
+class CloseScrollStrategy implements ScrollStrategy {
+  constructor (
+    private isActive: Ref<boolean>
+  ) {}
+
+  enable () {
+    document.addEventListener('scroll', this.onScroll.bind(this), { passive: true })
+  }
+
+  disable () {
+    document.removeEventListener('scroll', this.onScroll.bind(this))
+  }
+
+  private onScroll () {
+    this.isActive.value = false
+  }
+}
+
+class BlockScrollStrategy implements ScrollStrategy {
+  private initialOverflow = ''
+
+  enable () {
+    const el = document.documentElement
+    this.initialOverflow = el.style.overflowY
+
+    el.style.paddingRight = convertToUnit(window.innerWidth - el.offsetWidth)
+
+    el.style.overflowY = 'hidden'
+  }
+
+  disable () {
+    document.documentElement.style.overflowY = this.initialOverflow
+
+    document.documentElement.style.paddingRight = ''
+  }
+}
+
+export default defineComponent({
+  name: 'VOverlay',
+
+  directives: { ClickOutside },
+
+  inheritAttrs: false,
+
+  props: makeProps({
+    absolute: Boolean,
+    attach: {
+      type: [Boolean, String, Element] as PropType<boolean | string | Element>,
+      default: 'body',
+    },
+    eager: Boolean,
+    noClickAnimation: Boolean,
+    persistent: Boolean,
+    modelValue: Boolean,
+    positionStrategy: {
+      type: String as PropType<typeof positionStrategies[number]>,
+      default: 'global',
+      validator: (val: any) => positionStrategies.includes(val),
+    },
+    scrollStrategy: {
+      type: String as PropType<typeof scrollStrategies[number]>,
+      default: 'block',
+      validator: (val: any) => scrollStrategies.includes(val),
+    },
+    origin: [String, Object] as Prop<string | Element>,
+    scrim: {
+      type: [String, Boolean],
+      default: true,
+    },
+    ...makeTransitionProps(),
+  }),
+
+  emits: ['click:outside'],
+
+  setup (props, { slots, attrs, emit }) {
+    const isActive = useProxiedModel(props, 'modelValue')
+    const { teleportTarget } = useTeleport(toRef(props, 'attach'))
+    const { themeClasses } = provideTheme()
```

Comment:
use `useTheme`

Reply:
Overlay is moved to the root of body so it won't inherit anything without this.

Reply:
Hm. Am I misunderstanding this?

> This also means that injections from a parent component work as expected, and that the child component will be nested below the parent component in the Vue Devtools, instead of being placed where the actual content moved to.

https://v3.vuejs.org/guide/teleport.html#using-with-vue-components

Reply:
Misunderstood what useTheme did